wasm/text/index_namespace.go

```diff
@@ -0,0 +1,177 @@
+package text
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/tetratelabs/wazero/wasm"
+)
+
+func newIndexNamespace() *indexNamespace {
+	return &indexNamespace{idToIdx: map[string]wasm.Index{}}
+}
+
+// indexNamespace contains the count in an index namespace and any association of symbolic IDs to numeric indices.
```

Comment:
this type I hope to be able to mostly re-use in order to handle function locals, which are the next complex thing to attack and why I spent several days preparing for it here.